internal/sqlsmith: TestGenerateParse failed

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3937220&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3937220&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
FROM
	defaultdb.public.seed_mr_table@[0] AS tab_29
WHERE
	tab_29._bool
ORDER BY
	tab_29._decimal ASC
LIMIT
	61:::INT8;

STMT: 30
UPDATE
	defaultdb.public.seed_mr_table AS tab_30
SET
	_timestamptz = tab_30._timestamptz
WHERE
	(tab_30._bool OR tab_30._bool)
LIMIT
	47:::INT8;

    panic.go:1038: invalid argument to Intn
        goroutine 837502 [running]:
        runtime/debug.Stack()
        	GOROOT/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x11f
        panic({0x43bb700, 0x6223a00})
        	GOROOT/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc001dba360, {0x6316f00, 0xc00007e048})
        	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x389
        panic({0x43bb700, 0x6223a00})
        	GOROOT/src/runtime/panic.go:1038 +0x215
        math/rand.(*Rand).Intn(0x455da20, 0xc00f44a960)
        	GOROOT/src/math/rand/rand.go:168 +0x65
        math/rand.Intn(...)
        	GOROOT/src/math/rand/rand.go:337
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlterDatabaseDropRegion(0xc00f44a960)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:446 +0x36
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlter(0xc00d716c40)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:83 +0x7e
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).makeStmt(0xc00d716c40)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/relational.go:21 +0x35
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0x628a6a0)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:158 +0x31
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc00a544680)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x786
        testing.tRunner(0xc00a544680, 0x527abf8)
        	GOROOT/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	GOROOT/src/testing/testing.go:1306 +0x35a
--- FAIL: TestGenerateParse (1.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3937219&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3937219&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
        	GOROOT/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x11f
        panic({0x452ab80, 0x8e41730})
        	GOROOT/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc002278240, {0x630c000, 0xc00007e048})
        	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x325
        panic({0x452ab80, 0x8e41730})
        	GOROOT/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).formatNodeOrHideConstants(0x8e4f860, {0x0, 0x0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/hide_constants.go:46 +0x192
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc00307caf0, {0x0, 0x0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:444 +0xd5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags({0x0, 0x0}, 0x48040, {0x0, 0x10, 0xc000a5e000})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:582 +0x69
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).docAsString(0x203000, {0x0, 0x0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:160 +0x35
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0x4b71cb2, {0x0, 0x0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:155 +0x65
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*OnJoinCond).doc(0xc005a0b310, 0x9257480)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:932 +0x85
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0x45e3a7, {0x7f5756eccab0, 0xc00045f6d0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*JoinTableExpr).doc(0xc0058cbc70, 0x6280ce0)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:907 +0xd5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc005a0b448, {0x6280ce0, 0xc0058cbc70})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.TableExprs.doc({0xc00045f700, 0x1, 0x2}, 0x9257480)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:272 +0xc5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*From).docRow(0xc002d93400, 0x2)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:592 +0x51
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).docTable(0xc002d93400, 0x6343290)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:576 +0x1c7
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).docTable(0xc0058cbcc0, 0x4570c00)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:548 +0xdf
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).doc(0x45ad0d, 0x9257480)
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:541 +0x1e
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc005a0b710, {0x6281480, 0xc0058cbcc0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Pretty(0x9257480, {0x6281480, 0xc0058cbcc0})
        	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:144 +0x28
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0x6285c40)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:167 +0x6f
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc000cf49c0)
        	github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x786
        testing.tRunner(0xc000cf49c0, 0x5272e40)
        	GOROOT/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	GOROOT/src/testing/testing.go:1306 +0x35a
--- FAIL: TestGenerateParse (0.55s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936224&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936224&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc00806e240, {0x62c6980, 0xc000074108})
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x389
        panic({0x44e0fc0, 0x8bf3720})
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).formatNodeOrHideConstants(0x8c01720, {0x0, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/hide_constants.go:46 +0x192
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc005d67810, {0x0, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:444 +0xd5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags({0x0, 0x0}, 0x48040, {0x0, 0x10, 0xc001b80000})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:582 +0x69
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).docAsString(0x20, {0x0, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:160 +0x35
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0x4b266b2, {0x0, 0x0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:155 +0x65
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*OnJoinCond).doc(0xc01127f220, 0x9008be0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:932 +0x85
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0x62369c0, {0x7f57955051e8, 0xc00eda5c30})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*JoinTableExpr).doc(0xc00d9b6500, 0x62350a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:907 +0xd5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc01127f398, {0x62350a0, 0xc00d9b6500})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*JoinTableExpr).doc(0xc00d9b6550, 0x62350a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:924 +0x21e
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc01127f448, {0x62350a0, 0xc00d9b6550})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.TableExprs.doc({0xc00eda5c60, 0x1, 0x1}, 0x9008be0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:272 +0xc5
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*From).docRow(0xc00e1c8fa0, 0x1)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:592 +0x51
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).docTable(0xc00e1c8fa0, 0x62fdc10)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:576 +0x1c7
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).docTable(0xc00d9b65a0, 0x45270c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:548 +0xdf
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).doc(0x44e56d, 0x9008be0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:541 +0x1e
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Doc(0xc01127f710, {0x6235840, 0xc00d9b65a0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:152 +0x42
        github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*PrettyCfg).Pretty(0x9008be0, {0x6235840, 0xc00d9b65a0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/pretty.go:144 +0x28
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0x623a000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:167 +0x6f
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc0054c71e0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x786
        testing.tRunner(0xc0054c71e0, 0x5227620)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- FAIL: TestGenerateParse (1.17s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936223&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936223&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
	RIGHT JOIN (VALUES ('!':::STRING), ('/':::STRING), ('?':::STRING), ('H':::STRING)) AS tab_96 (col_142) ON false
LIMIT
	68:::INT8;

STMT: 12
WITH
	with_18 (col_145)
		AS (SELECT * FROM (VALUES (3281322748:::OID), (0:::OID), (3231287264:::OID), (0:::OID)) AS tab_97 (col_145)),
	with_19 (col_146) AS (SELECT * FROM (VALUES (NULL), ('1987-10-16':::DATE)) AS tab_98 (col_146))
SELECT
	('01:04:34.283033':::TIME::TIME + '-72 years -2 mons -96 days -18:55:09.441786':::INTERVAL::INTERVAL)::TIME
		AS col_147;

STMT: 13
COMMIT TRANSACTION;

STMT: 14
COMMIT TRANSACTION;

    panic.go:1038: invalid argument to Intn
        goroutine 11647 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x11f
        panic({0x4360fa0, 0x61c9ec0})
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc0038b4240, {0x62bbc20, 0xc000074108})
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x325
        panic({0x4360fa0, 0x61c9ec0})
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        math/rand.(*Rand).Intn(0xc0047736d8, 0x3e67f4c)
        	/usr/local/go/src/math/rand/rand.go:168 +0x65
        math/rand.Intn(...)
        	/usr/local/go/src/math/rand/rand.go:337
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlterDatabaseDropRegion(0xc000f660f0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:446 +0x36
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlter(0xc00420ae00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:83 +0x7e
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).makeStmt(0xc00420ae00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/relational.go:21 +0x35
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0x6235740)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:158 +0x31
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc00298d6c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x786
        testing.tRunner(0xc00298d6c0, 0x521fa08)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- FAIL: TestGenerateParse (0.29s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936226&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936226&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
		NULL::GEOGRAPHY,
		3.0:::FLOAT8::FLOAT8,
		('::/0':::INET::INET >> '::8/0':::INET::INET)::BOOL::BOOL
	)::BOOL
GROUP BY
	cte_ref_15.col_224;

STMT: 47
SELECT
	tab_144.col_228 AS col_229, tab_143.col_227 AS col_230, 0:::INT8 AS col_231
FROM
	(VALUES (st_zmflag(NULL::GEOMETRY)::INT2), (7:::INT8)) AS tab_142 (col_226),
	(VALUES ('::3/0':::INET), ('::9/0':::INET)) AS tab_143 (col_227),
	(VALUES (NULL), (NULL), (NULL), (NULL), (NULL)) AS tab_144 (col_228)
WHERE
	true;

I211217 10:59:56.491849 13440 1@gossip/gossip.go:1488  [n1] 743  node has connected to cluster via gossip
I211217 10:59:56.492667 13440 kv/kvserver/stores.go:269  [n1] 744  wrote 0 node addresses to persistent storage
    panic.go:1047: invalid argument to Intn
        goroutine 13496 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x72
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x94
        panic({0x65d7f20, 0xae5fa10})
        	/usr/local/go/src/runtime/panic.go:1047 +0x266
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc001a96480, {0xaf52580, 0xc0001c2008})
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x49a
        panic({0x65d7f20, 0xae5fa10})
        	/usr/local/go/src/runtime/panic.go:1047 +0x266
        math/rand.(*Rand).Intn(0xc0028d83c0, 0x0)
        	/usr/local/go/src/math/rand/rand.go:168 +0x99
        math/rand.Intn(...)
        	/usr/local/go/src/math/rand/rand.go:337
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlterDatabaseAddRegion(0xc003b50270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:435 +0x58
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlter(0xc0028d83c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:83 +0xa8
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).makeStmt(0xc0028d83c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/relational.go:21 +0x56
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0xaecb420)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:158 +0x3d
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc001bf1a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x9db
        testing.tRunner(0xc001bf1a00, 0x74991a8)
        	/usr/local/go/src/testing/testing.go:1259 +0x230
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x727
--- FAIL: TestGenerateParse (2.21s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3941597&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3941597&tab=artifacts#/) on master @ [9557b421269800165eeb6c0091920a090008a616](https://github.com/cockroachdb/cockroach/commits/9557b421269800165eeb6c0091920a090008a616):


```
	NULL;

STMT: 88
WITH
	with_30 (col_193)
		AS (
			SELECT
				*
			FROM
				(VALUES (NULL), (NULL), (NULL), (st_box2dfromgeohash(NULL::STRING, 7:::INT8::INT8)::BOX2D))
					AS tab_110 (col_193)
		),
	with_31 (col_194) AS (SELECT * FROM (VALUES (2:::OID), (6:::OID), (NULL)) AS tab_111 (col_194))
SELECT
	9:::INT8 AS col_195;

STMT: 89
COMMIT TRANSACTION;

    panic.go:1038: invalid argument to Intn
        goroutine 11829 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x11f
        panic({0x4361fc0, 0x61cba80})
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc00151e360, {0x62bd7a0, 0xc0001b2000})
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:548 +0x325
        panic({0x4361fc0, 0x61cba80})
        	/usr/local/go/src/runtime/panic.go:1038 +0x215
        math/rand.(*Rand).Intn(0xc0039a36d8, 0x3e68e8c)
        	/usr/local/go/src/math/rand/rand.go:168 +0x65
        math/rand.Intn(...)
        	/usr/local/go/src/math/rand/rand.go:337
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlterDatabaseDropRegion(0xc004ced800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:446 +0x36
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.makeAlter(0xc0007428c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/alter.go:83 +0x7e
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).makeStmt(0xc0007428c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/relational.go:21 +0x35
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.(*Smither).Generate(0x6237340)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith.go:158 +0x31
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc00168e1a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:184 +0x786
        testing.tRunner(0xc00168e1a0, 0x5220d28)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
--- FAIL: TestGenerateParse (1.02s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/internal/sqlsmith/alter.go
+++ b/pkg/internal/sqlsmith/alter.go
@@ -423,6 +423,9 @@ func makeAlterLocality(s *Smither) (tree.Statement, bool) {
 		},
 	}
 	if localityLevel == tree.LocalityLevelTable {
+		if len(regions) == 0 {
+			return &tree.AlterDatabaseAddRegion{}, false
+		}
 		ast.Locality.TableRegion = tree.Name(regions[rand.Intn(len(regions))])
 	}
 	return ast, ok
@@ -430,6 +433,10 @@ func makeAlterLocality(s *Smither) (tree.Statement, bool) {
 
 func makeAlterDatabaseAddRegion(s *Smither) (tree.Statement, bool) {
 	regions := getClusterRegions(s)
+
+	if len(regions) == 0 {
+		return &tree.AlterDatabaseAddRegion{}, false
+	}
 
 	ast := &tree.AlterDatabaseAddRegion{
 		Region: tree.Name(regions[rand.Intn(len(regions))]),
@@ -441,6 +448,10 @@ func makeAlterDatabaseAddRegion(s *Smither) (tree.Statement, bool) {
 
 func makeAlterDatabaseDropRegion(s *Smither) (tree.Statement, bool) {
 	regions := getDatabaseRegions(s)
+
+	if len(regions) == 0 {
+		return &tree.AlterDatabaseDropRegion{}, false
+	}
 
 	ast := &tree.AlterDatabaseDropRegion{
 		Region: tree.Name(regions[rand.Intn(len(regions))]),

